engine/access/rest/rest_api.go

```diff
@@ -0,0 +1,88 @@
+package rest
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine/access/rest/generated"
+	"github.com/onflow/flow-go/engine/access/rpc/backend"
+	"github.com/onflow/flow-go/model/encoding"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// RestAPIHandler provides the implementation of each of the REST API
+type RestAPIHandler struct {
+	backend *backend.Backend
+	logger  zerolog.Logger
+	encoder encoding.Encoder
+}
+
+func NewRestAPIHandler(backend *backend.Backend, logger zerolog.Logger) *RestAPIHandler {
+	return &RestAPIHandler{
+		backend: backend,
+		logger:  logger,
+		encoder: encoding.DefaultEncoder, //use the default JSON encoder
+	}
+}
+
+func (restAPI *RestAPIHandler) BlocksIdGet(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	vars := mux.Vars(r)
+	idParam, ok := vars["id"]
+	if !ok {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	// gorilla mux retains opening and ending square brackets for ids
+	idParam = strings.TrimSuffix(idParam, "]")
+	idParam = strings.TrimPrefix(idParam, "[")
+
+	ids := strings.Split(idParam, ",")
+
+	blocks := make([]*generated.Block, len(ids))
+
+	for i, id := range ids {
+		flowID, err := flow.HexStringToIdentifier(id)
+		if err != nil {
+			restAPI.errorResponse(w, r, err)
+		}
```

Comment:
Here It would not be appropriate to return a generic `StatusInternalServerError`, because the error is on the user side (they provided a malformed Hex ID).